src/QMCApp/QMCMain.cpp

```diff
@@ -446,7 +446,18 @@ bool QMCMain::validateXML()
     }
     else if (cname == "hamiltonian")
     {
-      ham_pool_->put(cur);
+      bool batched;
+      // since driver version is an enum this bool should be set explicitly for each version.
+      switch (my_project_.getDriverVersion())
+      {
+      case ProjectData::DriverVersion::LEGACY:
+        batched = false;
+        break;
+      case ProjectData::DriverVersion::BATCH:
+        batched = true;
+        break;
+      }
+      ham_pool_->put(cur, batched);
```

Comment:
Could the following work?
```
ham_pool_->put(cur, my_project_.getDriverVersion() == ProjectData::DriverVersion::BATCH);
```
If it works, please change the following place as well.

Reply:
Done